src/hotspot/os/bsd/os_perf_bsd.cpp

```diff
@@ -52,17 +52,150 @@
 
 static const double NANOS_PER_SEC = 1000000000.0;
 
+#if !defined(__APPLE__)
+struct CPUTicks {
+  uint64_t usedTicks;
+  uint64_t totalTicks;
+};
+
+struct JVMTicks {
+  uint64_t userTicks;
+  uint64_t systemTicks;
+  CPUTicks cpuTicks;
+};
+
+struct PerfCounters {
```

Comment:
I'm curious as to understand why you chose to do these a C structs in a C++ class implementation.  I'll admit it's a little odd to me that we're doing this rather than having them be tiny C++ classes.  Also, I would suggest that instead of having a separate PerfCounters struct/class that we just make all of these member variables of CPUPerformance.

Reply:
This is a by-product of following the implementation style of the Linux implementation. Agree on all points about moving it to a more natural C++ based result.